# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1236684471 -3600
#      Tue Mar 10 12:27:51 2009 +0100
# Node ID 1e4b3149365a710ecd4d9d4298e217dffb173f76
# Parent  e7e928088e9005b27ee92f5d71f78f6fe47b67ef
allow size_equal called with 1 arg

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2009-03-10  Jaroslav Hajek  <highegg@gmail.com>
+
+	* data.cc (Fsize_equal): Allow single argument call.
+
 2009-03-09  Benjamin Lindner  <lindnerb@users.sourceforge.net>
 
 	* ls-ascii-helper.h, ls-ascii-helper.cc: New files.
 	* Makefile.in: Add them to the appropriate lists.
 	* load-save.cc (Fload): Open all files in binary mode.
 	* ov-range.cc (load_ascii): Explicitly handle CR and CRLF line endings.
 	* ov-fcn-handle.cc (load_ascii): Likewise.
 	* ov-fcn-inline.cc (load_ascii): Likewise.
diff --git a/src/data.cc b/src/data.cc
--- a/src/data.cc
+++ b/src/data.cc
@@ -2430,24 +2430,25 @@ returns the number of columns in the giv
   return retval;
 }
 
 DEFUN (size_equal, args, ,
    "-*- texinfo -*-\n\
 @deftypefn {Built-in Function} {} size_equal (@var{a}, @var{b}, @dots{})\n\
 Return true if the dimensions of all arguments agree.\n\
 Trailing singleton dimensions are ignored.\n\
+Called with a single argument, size_equal returns true.\n\
 @seealso{size, numel}\n\
 @end deftypefn")
 {
   octave_value retval;
 
   int nargin = args.length ();
 
-  if (nargin >= 2)
+  if (nargin >= 1)
     {
       retval = true;
 
       dim_vector a_dims = args(0).dims ();
       a_dims.chop_trailing_singletons ();
 
       for (int i = 1; i < nargin; ++i)
         {
